pyccl/tests/test_neutrinos.py

```diff
@@ -12,7 +12,8 @@
     [0.1, 0.8, 0.3],
     np.array([0.1, 0.8, 0.3])])
 def test_omnuh2_smoke(a, m):
-    om = ccl.Omega_nu_h2(a, m_nu=m)
+    with pytest.warns(ccl.CCLDeprecationWarning):
+        om = ccl.Omeganuh2(a, m_nu=m)
```

Comment:
Can we start doing this properly with omega_x?

Reply:
Yea this is already tested in `test_background.py`. In v3, we're going to delete those tests, or I can delete them now if you prefer.

Reply:
OK, good. Can we add just one test, for my own peace of mind, that verifies that we get the same answer with omega_x and with Omeganuh2?

Reply:
They both call the same C function, `ccl_Omeganuh2`:
```C
double ccl_omega_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int *status)
{
  // If massive neutrinos are present, compute the phase-space integral and
  // get OmegaNuh2. If not, set OmegaNuh2 to zero.
  double OmNuh2;
  if ((cosmo->params.N_nu_mass) > 0.0001) {
    // Call the massive neutrino density function just once at this redshift.
    OmNuh2 = ccl_Omeganuh2(a, cosmo->params.N_nu_mass, cosmo->params.m_nu,
                           cosmo->params.T_CMB, cosmo->params.T_ncdm, status);
  } else {
    OmNuh2 = 0.;
  }
```
but there is a slight difference. `ccl_omega_x` returns zero if `N_nu_mass <= 1e-4` whereas `ccl_Omeganuh2` returns zero if `N_nu_mass` is exactly zero. I have removed the check from `ccl_omega_x` in order to let `ccl_Omeganuh2` do the job, and now both calls should be identical, so I don't feel there is need of unit testing, since the input of `ccl_omega_x` is directly piped into `ccl_Omeganuh2`.